moz_security/sandboxes/heka/analysis/moz_security_cloudtrail.lua

```diff
@@ -0,0 +1,147 @@
+            { "userIdentity.sessionContext.attributes.mfaAuthenticated", "^false$" }
+        }
+    }
+}
+
+-- mapping of aws account ids to human-friendly names (optional)
+aws_account_mapping = {
+    {"5555555555", "dev"},
+    {"1234567890", "prod"}
+}
+
+
+-- module makes use of alert output and needs a valid alert configuration
+alert = {
+    modules = { }
+}
+```
+--]]
+
+require "cjson"
+
+local alert = require "heka.alert"
+
+local cfevents = read_config("events")
+local cfaccount_name_mapping = read_config("aws_account_mapping")
+
+local payloadfields = {
+    "awsRegion",
+    "eventID",
+    "eventName",
+    "eventSource",
+    "eventTime",
+    "eventType",
+    "eventVersion",
+    "recipientAccountId",
+    "requestID",
+    "requestParameters.resourceName",
+    "requestParameters.userName",
+    "responseElements.accessKey.accessKeyId",
+    "responseElements.accessKey.createDate",
+    "responseElements.accessKey.status",
+    "responseElements.accessKey.userName",
+    "sourceIPAddress",
+    "userAgent",
+    "userIdentity.accessKeyId",
+    "userIdentity.accountId",
+    "userIdentity.arn",
+    "userIdentity.invokedBy",
+    "userIdentity.principalId",
+    "userIdentity.sessionContext.attributes.creationDate",
+    "userIdentity.sessionContext.attributes.mfaAuthenticated",
+    "userIdentity.sessionContext.sessionIssuer.accountId",
+    "userIdentity.sessionContext.sessionIssuer.arn",
+    "userIdentity.sessionContext.sessionIssuer.principalId",
+    "userIdentity.sessionContext.sessionIssuer.type",
+    "userIdentity.sessionContext.sessionIssuer.userName",
+    "userIdentity.type",
+    "userIdentity.userName"
+}
+
+function genpayload()
+    local det = {}
+    for _,v in ipairs(payloadfields) do
+        local vk = string.format("Fields[%s]", v)
+        det[v] = read_message(vk)
+    end
+    return cjson.encode(det)
+end
+
+function get_account_name(account_id)
+    -- { {account_id, account_name}, ... }
+    if cfaccount_name_mapping ~= nil then
+      for _, v in ipairs(cfaccount_name_mapping) do
+          if v[1] == account_id then return v[2] end
+      end
+    end
+    return account_id
+end
+
+function process_message()
+    local event_id = read_message("Fields[eventID]")
+    local account_id = read_message("Fields[recipientAccountId]")
+
+    for _, event in ipairs(cfevents) do
+        local match_counter = 0
+
+        for _, field in ipairs(event['fields']) do
+            local v = read_message(string.format("Fields[%s]", field[1]))
+            if v ~= nil then
+                if string.match(v, field[2]) then
+                    match_counter = match_counter + 1
+                end
+            end
+        end
+
+        if match_counter == #event['fields'] then
+              local id = string.format("%s - %s", event["description"], event_id)
+              local s = string.format("%s in %s", event["description"], get_account_name(account_id))
```

Comment:
I think this is OK as is for now, it might be nice to have the ability to include information relevant to the event type in the subject (e.g., like a username or other) but we have all that in the payload so maybe not that important for the initial version